src/components/startup-form.tsx

```diff
@@ -0,0 +1,184 @@
+        resolver: zodResolver(formSchema),
+        defaultValues: {
+            year: "",
+            name: "",
+            description: "",
+            sector: "",
+            quote: "",
+            author: "",
+        },
+    });
+    useEffect(() => {
+        if (startupId) {
+            const startupRef = dbRef(database, `startups/${startupId}`);
+            get(startupRef).then((snapshot) => {
+                const startupData = snapshot.val();
+                if (startupData) {
+                    form.reset({
+                        year: startupData.year,
+                        name: startupData.name,
+                        sector: startupData.sector,
+                        description: startupData.description,
+                        quote: startupData.quote || "",
+                        author: startupData.author,
+                    });
+                }
+            });
+        }
+    }, [startupId, form]);
+    const onSubmit = async (values: FormValues) => {
+        try {
+            const startupData = {
+                year: values.year,
+                name: values.name,
+                sector: values.sector,
+                description: values.description,
+                quote: values.quote || "",
+                author: values.author,
+            };
+
+            if (startupId) {
+                const startupRef = dbRef(database, `startups/${startupId}`);
+                await set(startupRef, startupData);
+                toast.success("Startup actualizado satisfactoriamente!");
+            } else {
+                const newStartupRef = push(dbRef(database, "startups"));
+                await set(newStartupRef, startupData);
+                toast.success("Startup creado satisfactoriamente!");
+            }
+
+            form.reset();
+            router.push("/startups");
+
+        } catch (error) {
+            toast.error("Error al enviar el formulario");
+        }
+    };
+    return(
+        <>
+            <div className="m-9 p-9 border border-black rounded-lg custom-shadow overflow-hidden max-w-6xl mx-auto flex flex-col gap-6 min-h-3.5">
+                <Form {...form}>
+                    <div className="flex flex-col md:flex-row gap-6">
+                        <div className="flex-1 flex flex-col gap-6">
+                            <FormField control={form.control} name="year" render={({ field }) => (
+                                <FormItem>
+                                    <FormLabel>Año</FormLabel>
+                                    <FormControl>
+                                        <Input placeholder="2024" {...field} />
+                                    </FormControl>
+                                    <FormMessage />
+                                </FormItem>
+                            )}
+                            />
+                            <FormField control={form.control} name="name" render={({ field }) => (
```

Comment:
you can create a component called TextField.tsx:
Example:

import { type FC } from 'react';
import { type Control, type FieldValues } from 'react-hook-form';
import {  FormControl, FormField, FormItem, FormLabel } from '@/components/ui/form';
import { Input } from '@/components/ui/input';
import { PropsWithClassName } from '@/shared/types';


type TextFieldType<T extends FieldValues> = {
    control: Control<T>;
    fieldName: keyof T;
    label: string;
    placeholder: string;
} & PropsWithClassName

export const TextField: FC<TextFieldType<any>> = ({ control, fieldName, label, placeholder, className }) => {
	return (
		<FormField
			control={control}
			name={fieldName as string}
			render={({ field }) => (
				<FormItem>
					<FormLabel>{label}</FormLabel>
					<FormControl>
						<Input placeholder={placeholder} {...field}  className={className}/>
					</FormControl>
				</FormItem>
			)}
		/>
	);
};


If you do it it will be easier in the future if you need to change text validations or changes on styles